arangod/Agency/Node.h

```diff
@@ -75,6 +75,60 @@ typedef std::chrono::steady_clock::time_point SteadyTimePoint;
 
 class Store;
 
+class SmallBuffer {
+  uint8_t* _start;
+  size_t _size;
+ public:
+  SmallBuffer() : _start(nullptr), _size(0) {}
+  SmallBuffer(size_t size) : _start(new uint8_t[size]), _size(size) {}
```

Comment:
Do we want to handle the case of `size == 0` here?
Not sure if it is really worth it.
```suggestion
  explicit SmallBuffer(size_t size) : SmallBuffer() {
    if (size > 0) {
      _start = new uint8_t[size];
      _size = size;
    }
  }
```